### PR TITLE
Eagerly validate created/updated docker images

### LINKED ISSUE #549: Eagerly validate custom-provided docker images
## Tell us about the problem you're trying to solve
When I create a custom source or update the version of a connector's image, I want to know if the provided docker image is invalid eagerly. 

## Describe the solution you’d like
I want to the API to return a failure code at creation/update time notifying me if there are any issues.

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Adds eager validation of docker images supplied when creating/updating sources/destinations. 

## How
Runs a `getSpec` operation on the provided docker image in the relevant endpoints. 

## Reading order
1. `SourcesHandler.java`
1. `SchedulerHandler.java`
1. `DestinationsHandler.java`
1. `SourcesHandlerTest.java`
1. `SchedulerHandlerTest.java`
1. `DestinationsHandlerTest.java`

### CODE REVIEW

File: airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java [hunk 1]
```diff
@@ -153,41 +153,42 @@ public SourceSpecificationRead getSourceSpecification(SourceIdRequestBody source
...
         .build());
 
-    final ConnectorSpecification spec = job.getOutput().orElseThrow().getGetSpec().getSpecification();
     return new SourceSpecificationRead()
         .connectionSpecification(spec.getConnectionSpecification())
         .documentationUrl(spec.getDocumentationUrl().toString())
         .sourceId(sourceId);
   }
 
+  ConnectorSpecification getConnectorSpecification(String imageName) throws IOException {
```
Comment (michel-tricot) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java:170]: why is this visible?

File: airbyte-server/src/main/java/io/airbyte/server/handlers/SourcesHandler.java [hunk 2]
```diff
@@ -96,6 +106,17 @@ public SourceRead updateSource(SourceUpdate sourceUpdate) throws ConfigNotFoundE
     return buildSourceRead(newSource);
   }
 
+  private void assertDockerImageIsValidIntegration(String dockerRepo, String tag) {
```
Comment (michel-tricot) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/SourcesHandler.java:109]: This code is repeted

**Review by michel-tricot — APPROVED** [review]

File: airbyte-server/src/test/java/io/airbyte/server/handlers/SourcesHandlerTest.java [hunk 3]
```diff
@@ -140,17 +144,61 @@ void testCreateSource() throws URISyntaxException, ConfigNotFoundException, IOEx
...
+  }
+
+  @Test
+  void testUpdateInvalidSourceFails() throws IOException, JsonValidationException, ConfigNotFoundException {
+    final String newDockerTag = "newtag123";
+    final String newDockerImage = DockerUtils.getTaggedImageName(source.getDockerRepository(), newDockerTag);
+    when(schedulerHandler.getConnectorSpecification(newDockerImage)).thenThrow(new IllegalArgumentException("invalid image"));
+    when(configRepository.getStandardSource(source.getSourceId())).thenReturn(source);
+
+    try {
```
Comment (michel-tricot) [inline airbyte-server/src/test/java/io/airbyte/server/handlers/SourcesHandlerTest.java:181]: you can use `assertThrow`

File: airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java (hunk 1 above)
Comment (sherifnada) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java:170]: we need a way to get the spec using only an imageName. The other methods (getSourceSpec/destinationSpec) require a source ID or destination ID. This is the method I'm exposing to allow ID-less retrieval of specs. Do you see an issue with this pattern?

**Review by cgardens — APPROVED** [review]

File: airbyte-server/src/main/java/io/airbyte/server/handlers/SourcesHandler.java (hunk 2 above)
Comment (sherifnada) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/SourcesHandler.java:109]: Good idea, i was just thinking of how to get around this.

Comment (sherifnada) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/SourcesHandler.java]: Added an image validator and deduped tests

**Pull request merged**